app/_kong_plugins/openid-connect/index.md

```diff
@@ -37,6 +35,541 @@ search_aliases:
+    deactivate kong
+    idp->>idp: authenticate client <br/>and introspect access token
+    activate kong
+    idp->>kong: return introspection response
+    deactivate idp
+    kong->>kong: verify introspection response
+    activate httpbin
+    kong->>httpbin: request with <br/>access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+### JWT access token authentication flow
+
+For legacy reasons, the stateless `JWT Access Token` authentication is named `bearer` 
+(see [`config.auth_methods`](/plugins/openid-connect/reference/)). Stateless authentication means
+that the signature verification uses the identity provider to publish public keys and the standard
+claims verification (such as `exp` or expiry). The client may receive the token directly
+from the identity provider or by other means.
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant httpbin as Upstream <br>(upstream service,<br> e.g. httpbin)
+    activate client
+    activate kong
+    client->>kong: service with<br>access token
+    deactivate client
+    kong->>kong: load access token
+    kong->>kong: verify signature
+    kong->>kong: verify claims
+    activate httpbin
+    kong->>httpbin: request with<br>access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+### Kong OAuth token auth flow
+
+The OpenID Connect plugin can verify the tokens issued by the [OAuth 2.0 plugin](/plugins/oauth2/).
+This is very similar to third party identity provider issued [JWT access token authentication](#jwt-access-token-authentication-flow)
+or [introspection authentication](#introspection-authentication-flow):
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant httpbin as Upstream <br>(upstream service,<br> e.g. httpbin)
+    activate client
+    activate kong
+    client->>kong: service with<br>access token
+    deactivate client
+    kong->>kong: load access token
+    kong->>kong: verify kong<br>oauth token
+    activate httpbin
+    kong->>httpbin: request with<br>access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+
+### Password grant workflow
+
+Password grant is a **legacy** authentication grant. This is a less secure way of
+authenticating end users than the authorization code flow, because, for example,
+the passwords are shared with third parties.
```

Comment:
```suggestion
Password grant is a **legacy** authentication grant. This is a less secure way of authenticating end users than the authorization code flow, because, for example, the passwords are shared with third parties.
```